packages/graphic-walker/src/components/pivotTable/metricTable.tsx

```diff
@@ -1,22 +1,47 @@
 import React from 'react';
-import { IField, IRow } from '../../interfaces';
+import { useMemo } from "react";
+import { IField, IRow, IVisualConfig } from '../../interfaces';
 import { getMeaAggKey } from '../../utils';
+import { format } from "d3-format";
 
 interface MetricTableProps {
     matrix: any[][];
     meaInRows: IField[];
     meaInColumns: IField[];
+    numberFormat: IVisualConfig['format']
 }
 
-function getCellData (cell: IRow, measure: IField) {
+function getCellData (cell: IRow, measure: IField, formatter: (value: unknown) => string) {
+
     const meaKey = getMeaAggKey(measure.fid, measure.aggName);
     if (cell[meaKey] === undefined) {
         return '--';
     }
-    return cell[meaKey];
+    const formattedValue = formatter(cell[meaKey]);
+    return formattedValue;
 }
 const MetricTable: React.FC<MetricTableProps> = React.memo((props) => {
-    const { matrix, meaInRows, meaInColumns } = props;
+    const { matrix, meaInRows, meaInColumns, numberFormat } = props;
+
+    const numberFormatter = useMemo<(value: unknown) => string>(() => {
```

Comment:
There may be non-number content in the table, so when the value is not a number, it should return the value itself.
In addition, the function appears to be a bit bloated, so it is recommended to write it in the following form:
```
const numberFormatter = useMemo<(value: unknown) => string>(() => {
    const numberFormatter = numberFormat ? format(numberFormat) : (v: number) => v.toLocaleString();
    return (value: unknown) => {
        if (typeof value !== "number") {
            return `${value}`;
        }
        return numberFormatter(value);
    };
}, [numberFormat]);
 ```

Reply:
Thanks a lot @islxyqwe for reviewing the PR, I have made the changes which you have suggested above, let me know if anything else needs to be changed in this